### DISCUSSION TITLE
How to install Harvest in kubernetes

### DISCUSSION THREAD

**Discussion by eselvam** [NetApp/harvest#827] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
We plan to install Harvest in kubernetes. Kindly help us  here with detailed instruction

**Describe the solution you'd like**
We need clear document on how to install Harvest in kubernetes

**Describe alternatives you've considered**
If k8 is not feasible for porting Harvest, we have to look for physical or virtual vm

**Additional context**
Harvest on K8

**Comment by rahulguptajss** [comment, marked as answer]
@eselvam does https://github.com/NetApp/harvest/tree/main/docker/k8 help?

**Comment by eselvam** [comment]
It has:

    bin/harvest generate docker full --port --output harvest-compose.yml
    kompose convert --file harvest-compose.yml --file prom-stack.yml --out kub.yaml --volumes hostPath
    kubectl apply --filename kub.yaml

do we still need the harvest need to be installed to generate yaml file for k8.
we use trident for pv and pvc  the --volumes has only hostPath. is it support pvc?
what about ingress? how the service are created? no such options. It is very basic instruction.

Shall you please share the detailed info.

**Comment by rahulguptajss** [comment]
@eselvam 

1. Harvest also ships in a tar file. Does that help in running harvest commands without installing it?
2. Harvest needs few configurations as input before pods can be started. This can be achieved with configmap or pv. You may have to use some sort of init scripts to copy these configurations into pv before pod starts.

Below are the configurations: (https://github.com/NetApp/harvest/blob/main/docker/onePollerPerContainer/docker-compose.tmpl#L30-L32)
- harvest configuration file (harvest.yaml)
- conf directory (conf folder).  [If you have no changes in this directory then you can ignore this volume. Harvest docker image has this folder]
- cert directory (cert folder) [If you use cert based authetication then this configuration is needed else can be ignored]

**ConfigMap:**
To create configMap for a folder, you can refer https://stackoverflow.com/questions/55790144/create-configmaps-from-files-recursively/59670136#59670136

Generate k8 config as below

```
bin/harvest generate docker --port --output harvest-compose.yml
kompose convert --file harvest-compose.yml --out kub.yaml --volumes configMap
```

**PV:**

PV needs to have all relevant volume configuration data copied before starting pods.

```
bin/harvest generate docker --port --output harvest-compose.yml
kompose convert --file harvest-compose.yml --out kub.yaml --volumes persistentVolumeClaim
```

kompose command above will generate service configuration as well.
